lib/vagrant/vagrantfile.rb

```diff
@@ -194,6 +195,11 @@ def machine_config(name, provider, boxes)
           original_box = config.vm.box
           load_box_proc.call
         end
+
+        if original_version != config.vm.box_version
+          original_version = config.vm.box_version
+          load_box_proc.call
+        end
```

Comment:
Lets make this a single conditional instead of broken into two:

```ruby
if original_box != config.vm.box || original_version != config.vm.box_version
  original_box = config.vm.box
  original_version = config.vm.box_version
  load_box_proc.call
end
```